File: hw1/hw1/hw1/Calculator.cs
namespace hw1
{
    public class Calculator
    {
        public static double Calculate(double firstarg, string operation, double secondarg)
        {
            double result = operation switch
            {
                "+" => firstarg + secondarg,
                "-" => firstarg - secondarg,
                "*" => firstarg * secondarg,
                "/" => (secondarg == 0) ? 0 : firstarg / secondarg,
                _ => 0
            };
            return result;
        }
    }
}

File: hw5/hw5/Calculator.fs
module hw5.Calculator
open hw5
open Parser
    
    let Calculate (arg1 : decimal) (operation) (arg2 : decimal) =
        if arg2 = (decimal) 0  then
            Failure "Divide by zero is not allowed."
        else
            let result =
                match operation with
                |  "+" ->   (arg1 + arg2) 
                |  "-" ->   (arg1 - arg2)
                |  "*" ->   (arg1 * arg2)
                |  "/" ->   (arg1 / arg2)
            Success result